### PR TITLE
Fix goroutine leak when using  `multiResizeQueue` and `spdy` connection leak

### PR DESCRIPTION
**Pull request by tigrato**
This PR fixes multiple goroutine and memory leaks when interactive sessions are used. 

- When the session terminates, the `multiResizeQueue` never returns, and the resize stream goroutine blocks. 
- A goroutine leak exited when the server received resizing events after the connection terminated - this happens with fast exec requests.
- A memory leak existed when users tried to leave the session after the `session.tracker` was closed. 

This PR also releases the connection monitor earlier. When the server is under heavy load, it might take a while for the connection to return an `EOF` - which triggers the service monitor automatic release - and the service monitor resources were leaking until the server resumed normal operation.

It also fixes reloads when new parties join and leave the `multiResizeQueue`.

### CODE REVIEW

File: lib/kube/proxy/sess.go [hunk 1]
```diff
@@ -165,34 +165,54 @@ func (p *kubeProxyClientStreams) sendStatus(err error) error {
...
+	r.reloadCtx, r.reloadCancel = context.WithCancel(r.parentCtx)
+	r.cases = make([]reflect.SelectCase, len(r.queues)+1)
+	r.cases[0] = reflect.SelectCase{
+		Dir:  reflect.SelectRecv,
+		Chan: reflect.ValueOf(r.reloadCtx.Done()),
+	}
+	i := 1
 	for _, queue := range r.queues {
-		r.cases = append(r.cases, reflect.SelectCase{
+		r.cases[i] = reflect.SelectCase{
```
Comment (LKozlowski) [inline lib/kube/proxy/sess.go:204]:
```suggestion
		r.cases[i+1] = reflect.SelectCase{
```
Then you could drop the `i` variable that you create before the loop, change index variable in the loop from `_` to `i` and then remove the increment operation at the bottom. 

That way we have less code to worry about 😄

**Review by LKozlowski — APPROVED** [review]

File: lib/kube/proxy/sess.go (hunk 1 above)
Comment (tigrato) [reply lib/kube/proxy/sess.go:204]: `r.queues` is a map whose key is a string.

Comment (LKozlowski) [reply lib/kube/proxy/sess.go:204]: Ahh, sorry, my bad!

File: lib/kube/proxy/sess.go [hunk 2]
```diff
@@ -424,16 +442,19 @@ func (s *session) disconnectPartyOnErr(idString string, err error) {
...
+		s.log.WithError(uuidParseErr).Errorf("Unable to decode %q into a UUID.", idString)
 		return
 	}
 
-	if err = s.leave(id); err != nil {
-		s.log.Errorf("Failed to disconnect party %v from the session: %v.", idString, err)
+	wasConnected, leveErr := s.leave(id)
+	if leveErr != nil {
+		s.log.WithError(leveErr).Errorf("Failed to disconnect party %v from the session.", idString)
+	}
```
Comment (LKozlowski) [inline lib/kube/proxy/sess.go:454]:
```suggestion
	wasConnected, leaveErr := s.leave(id)
	if leaveErr != nil {
		s.log.WithError(leaveErr).Errorf("Failed to disconnect party %v from the session.", idString)
	}
```

File: lib/kube/proxy/sess.go [hunk 3]
```diff
@@ -941,28 +971,30 @@ func (s *session) BroadcastMessage(format string, args ...interface{}) {
...
 	stringID := id.String()
 	party := s.parties[id]
 
 	if party == nil {
-		return nil
+		return false, nil
 	}
-
+	// Waits until the function execution ends to release the parties waitgroup.
+	// It used to prevent the session to terminate the events emitter before
```
Comment (LKozlowski) [inline lib/kube/proxy/sess.go:995]:
```suggestion
	// It's used to prevent the session to terminate the events emitter before
```

**Review by r0mant — APPROVED** [review]

**Pull request merged**